### PR TITLE
feat: init renderer server

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

**Comment by Brooooooklyn** [comment]
* **#8089** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8089?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8088** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8088?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7794** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7794?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Brooooooklyn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/core/doc-renderer/service.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    const content = parseWorkspaceDoc(doc);
+
+    if (content) {
+      await this.cache.set(cacheKey, content);
+    }
+
+    return content;
+  }
+
+  async markDocContentCacheStale(workspaceId: string, guid: string) {
```
Comment (EYHN) [inline packages/backend/server/src/core/doc-renderer/service.ts]: never used?

File: packages/backend/server/package.json [hunk 2]
```diff
@@ -94,6 +95,7 @@
     "ts-node": "^10.9.2",
     "typescript": "^5.4.5",
     "ws": "^8.16.0",
+    "xss": "^1.0.15",
```
Comment (darkskygit) [inline packages/backend/server/package.json]: not used dep?

File: packages/backend/server/src/core/auth/service.ts [hunk 3]
```diff
@@ -47,10 +47,12 @@ export class AuthService implements OnApplicationBootstrap {
...
     private readonly config: Config,
     private readonly db: PrismaClient,
-    private readonly mailer: MailService,
     private readonly feature: FeatureManagementService,
     private readonly quota: QuotaService,
-    private readonly user: UserService
+    private readonly user: UserService,
+    // mark the mailer as optional because the mailer is not available if app flavor is renderer
+    // and we don't mark the mailer as `mailer?: MailService` because the mailer code path won't be accessed in the renderer app
+    @Optional() private readonly mailer: MailService
```
Comment (forehalo) [inline packages/backend/server/src/core/auth/service.ts]: mailer service handled it already

**Review by forehalo — APPROVED** [review]

**Review by EYHN — APPROVED** [review]

File: packages/backend/server/src/core/selfhost/index.ts [hunk 4]
```diff
@@ -92,13 +95,13 @@ export class SelfhostModule implements OnModuleInit {
     });
     app.use(
       basePath,
-      serveStatic(staticPath, {
+      serveStatic(join(staticPath, 'selfhost'), {
```
Comment (bytemain) [inline packages/backend/server/src/core/selfhost/index.ts:98]: 这里没有考虑到 mobile => 'static/mobile/selfhost' 这个吧，

这里修了之后， render 那边要改一下返回的静态资源拼接地址：
image


也有人提了 issue 了: https://github.com/toeverything/AFFiNE/issues/8338
目前手机端没法访问 self host 的网站。

**Pull request merged**